### PR TITLE
feat: add state for global mode

### PR DESCRIPTION
**Pull request by ImCesar**
# Details
Global state can be activated if the CLI is passed the --global flag or if there is a lack of an active project. Added the graphql state to satisfy those conditions and added GlobalPage to App if the app is in global mode.

# How to Test
## Global Mode
At top level of repo run 

`LAUNCHPAD=1 yarn dev:watch --global`

or run `LAUNCHPAD=1 yarn dev:watch` with the lack of a project

image

### Provide Project
inside `packages/launchpad`  run 

`yarn cypress:open`

image

### CODE REVIEW

File: packages/server/lib/util/args.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const nestedObjectsInCurlyBracesRe = /\{(.+?)\}/g
 const nestedArraysInSquareBracketsRe = /\[(.+?)\]/g
 const everythingAfterFirstEqualRe = /=(.*)/
 
-const allowList = 'appPath apiKey browser ci ciBuildId clearLogs config configFile cwd env execPath exit exitWithCode group headed inspectBrk key logs mode outputPath parallel ping port project proxySource quiet record reporter reporterOptions returnPkg runMode runProject smokeTest spec tag updating version testingType'.split(' ')
+const allowList = 'appPath apiKey browser ci ciBuildId clearLogs config configFile cwd env execPath exit exitWithCode global group headed inspectBrk key logs mode outputPath parallel ping port project proxySource quiet record reporter reporterOptions returnPkg runMode runProject smokeTest spec tag updating version testingType'.split(' ')
```
Comment (JessicaSachs) [inline packages/server/lib/util/args.js:16]: Hmm. Why is this line necessary? I see that we support `--global` already.

File: packages/launchpad/src/App.vue [hunk 2]
```diff
@@ -15,6 +20,7 @@ import { gql, useQuery } from '@urql/vue'
 import Wizard from "./setup/Wizard.vue"
 import HeaderBar from './layouts/HeaderBar.vue'
 import { AppQueryDocument } from './generated/graphql'
+import GlobalPage from './global/GlobalPage.vue'
```
Comment (JessicaSachs) [inline packages/launchpad/src/App.vue:23]:
```suggestion
import GlobalEmpty from './global/GlobalEmpty.vue'
```

File: packages/launchpad/src/App.vue [hunk 3]
```diff
@@ -4,8 +4,13 @@
   </div>
 
   <div v-else class="bg-white h-full">
-    <HeaderBar :gql="query.data.value.app" />
-    <Wizard :query="query.data.value" />
+    <div v-if="app?.isInGlobalMode">
+      <GlobalPage />
```
Comment (JessicaSachs) [inline packages/launchpad/src/App.vue:8]:
```suggestion
      <GlobalEmpty />
```

File: packages/types/src/server.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ export interface LaunchArgs {
   cwd: string
   browser: Browser
   configFile?: string
+  global?: boolean
```
Comment (JessicaSachs) [inline packages/types/src/server.ts:17]:
```suggestion
  // Global mode is triggered by CLI via `--global` or when there is no `projectRoot` (essentially when the Cypress Config file can't be found)
  global?: boolean 
```

**Review by JessicaSachs — APPROVED** [review]
One quick question, some comments, and switching to `GlobalEmpty` instead of `GlobalPage` until the story for fetching the projects is completed.

File: packages/server/lib/util/args.js (hunk 1 above)
Comment (ImCesar) [reply packages/server/lib/util/args.js:16]: When it parses the args to turn into an array these are the only args that will get parsed into the array. Since global was missing from here it was never getting outputted in the parsed args

**Review by JessicaSachs — APPROVED** [review]

File: packages/graphql/src/entities/App.ts [hunk 5]
```diff
@@ -18,6 +18,16 @@ export class App {
     return true
   }
 
+  @nxs.field.boolean({
```
Comment (lmiller1990) [inline packages/graphql/src/entities/App.ts:21]: I think this should be required - eg `true` or `false` (cannot be `undefined` - we are in global mode, or we are not).

You can add `nonNull`:

```ts
@nxs.field.nonNull.boolean
```

File: packages/graphql/src/entities/App.ts [hunk 6]
```diff
@@ -18,6 +18,16 @@ export class App {
...
+    description: 'Whether the app is in global mode or not',
+  })
+  get isInGlobalMode (): NxsResult<'App', 'isInGlobalMode'> {
+    const hasGlobalModeArg = this.ctx.launchArgs.global ?? false
+    const isMissingActiveProject = !this.ctx.activeProject?.id
```
Comment (lmiller1990) [inline packages/graphql/src/entities/App.ts:26]: I think w can just check for `activeProject` as null - I don't think `?.id` adds anything.

File: packages/launchpad/src/App.vue [hunk 7]
```diff
@@ -4,8 +4,13 @@
   </div>
 
   <div v-else class="bg-white h-full">
-    <HeaderBar :gql="query.data.value.app" />
-    <Wizard :query="query.data.value" />
+    <div v-if="app?.isInGlobalMode">
```
Comment (lmiller1990) [inline packages/launchpad/src/App.vue:7]: Since `App` is non nullable, and we have the check for `query.data.value` in the `v-if`, I think we can just do:

```suggestion
    <div v-if="query.data.value.app.isInGlobalMode">
```

Then we won't need the additional `computed` property, either.

File: packages/types/src/server.ts [hunk 8]
```diff
@@ -14,6 +14,8 @@ export interface LaunchArgs {
   cwd: string
   browser: Browser
   configFile?: string
+  // Global mode is triggered by CLI via `--global` or when there is no `projectRoot` (essentially when the Cypress Config file can't be found)
```
Comment (lmiller1990) [inline packages/types/src/server.ts:17]: I don't think this should be nullable - either `true` or `false` (making in nullable doesn't really add any value. We'd assume `undefined` is `false`, so I think it's better be explicit.

File: scripts/watch.js [hunk 9]
```diff
@@ -22,7 +22,7 @@ function runServer () {
     child.removeAllListeners()
   }
 
-  child = childProcess.fork(path.join(__dirname, 'start.js'), ['--devWatch'], {
+  child = childProcess.fork(path.join(__dirname, 'start.js'), [...process.argv, '--devWatch'], {
```
Comment (lmiller1990) [inline scripts/watch.js:25]: Why do we need this?

**Review by lmiller1990 — COMMENTED** [review]
I left some comments/questions/recommendations!

**Comment by lmiller1990** [comment]
https://github.com/cypress-io/cypress/pull/18073 got merged so you may need a quick rebase/conflict resolution.

File: scripts/watch.js (hunk 9 above)
Comment (tgriesser) [reply scripts/watch.js:25]: b/c we're currently not passing along the CLI args to the `start.js` script:

```
../../scripts/start.js --component-testing --project ${PWD} # gets proper argv

../../scripts/watch.js --component-testing --project ${PWD} # only gets --devWatch
```

**Review by tgriesser — APPROVED** [review]
